clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -101,7 +101,10 @@
     DESCRIBE_CLUSTER(ApiMessageType.DESCRIBE_CLUSTER),
     DESCRIBE_PRODUCERS(ApiMessageType.DESCRIBE_PRODUCERS),
     BROKER_REGISTRATION(ApiMessageType.BROKER_REGISTRATION, true, RecordBatch.MAGIC_VALUE_V0, false, true),
-    BROKER_HEARTBEAT(ApiMessageType.BROKER_HEARTBEAT, true, RecordBatch.MAGIC_VALUE_V0, false, true);
+    BROKER_HEARTBEAT(ApiMessageType.BROKER_HEARTBEAT, true, RecordBatch.MAGIC_VALUE_V0, false, true),
+
+    // Once we have the controller integration for supporting broker decommissioning, we will support forwarding from the broker
```

Comment:
Maybe emphasize that we have temporarily marked this as "controller-only" as a short-term workaround to avoid exposing it on zk-based brokers.

Reply:
Ack.